File: examples/go/eks_helpers_example.go
package main

import (
	"github.com/castai/pulumi-castai/sdk/go/castai" // Assuming main package
	"github.com/pulumi/pulumi-aws/sdk/v6/go/aws"    // To get caller identity
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

// runEksHelpersExample demonstrates EKS helper resources.
// Renamed from main to avoid conflicts.
func runEksHelpersExample(ctx *pulumi.Context) error {
	// Placeholder values
	clusterRegion := "us-west-2"
	clusterName := "my-production-eks"

	// Get current AWS Account ID
	callerIdentity, err := aws.GetCallerIdentity(ctx, nil)
	if err != nil {
		return err
	}
	accountId := callerIdentity.AccountId

	// Get the CAST AI Cluster ID for the EKS cluster
	castaiClusterId, err := castai.NewEksClusterid(ctx, "example-go-eks-clusterid", &castai.EksClusteridArgs{
		AccountId:   pulumi.String(accountId),
		Region:      pulumi.String(clusterRegion),
		ClusterName: pulumi.String(clusterName),
	})
	if err != nil {
		return err
	}

	// Get the required User ARN using the CAST AI Cluster ID
	castaiUserArn, err := castai.NewEksUserArn(ctx, "example-go-eks-userarn", &castai.EksUserArnArgs{
		ClusterId: castaiClusterId.ID(), // Use the output from the previous resource
	})
	if err != nil {
		return err
	}

	// Export the Cluster ID and User ARN
	ctx.Export("eks_castai_go_cluster_id", castaiClusterId.ID())
	ctx.Export("eks_castai_go_user_arn", castaiUserArn.Arn) // Assuming the output property is 'Arn'

	return nil
}

/*
// Example of how you might call this from your main function:
func main() {
    pulumi.Run(func(ctx *pulumi.Context) error {
        if err := runEksHelpersExample(ctx); err != nil {
            return err
        }
        // ... potentially call other example functions ...
        return nil
    })
}
*/

File: examples/typescript/eks_helpers_example.ts
import * as pulumi from "@pulumi/pulumi";
import * as castai from "@castai/pulumi-castai"; // Use your actual package name
import * as aws from "@pulumi/aws"; // To get caller identity

// Placeholder values
const clusterRegion = "us-west-2";
const clusterName = "my-production-eks";

// Get current AWS Account ID
const callerIdentity = aws.getCallerIdentity({});
const accountId = callerIdentity.then(id => id.accountId);

// Get the CAST AI Cluster ID for the EKS cluster
const castaiClusterId = new castai.EksClusterid("example-ts-eks-clusterid", {
    accountId: accountId,
    region: clusterRegion,
    clusterName: clusterName,
});

// Get the required User ARN using the CAST AI Cluster ID
const castaiUserArn = new castai.EksUserArn("example-ts-eks-userarn", {
    clusterId: castaiClusterId.id, // Use the output from the previous resource
});

// Export the Cluster ID and User ARN
export const eksCastaiClusterId = castaiClusterId.id;
export const eksCastaiUserArn = castaiUserArn.arn; // Assuming the output property is 'arn' 